src/ty/mod.rs

```diff
@@ -68,22 +68,38 @@ pub struct Type<T: Form = MetaForm> {
         feature = "serde",
         serde(skip_serializing_if = "Path::is_empty", default)
     )]
-    path: Path<T>,
+    pub(crate) path: Path<T>,
     /// The generic type parameters of the type in use. Empty for non generic types
     #[cfg_attr(
         feature = "serde",
         serde(rename = "params", skip_serializing_if = "Vec::is_empty", default)
     )]
-    type_params: Vec<TypeParameter<T>>,
+    pub(crate) type_params: Vec<TypeParameter<T>>,
     /// The actual type definition
     #[cfg_attr(feature = "serde", serde(rename = "def"))]
-    type_def: TypeDef<T>,
+    pub(crate) type_def: TypeDef<T>,
     /// Documentation
     #[cfg_attr(
         feature = "serde",
         serde(skip_serializing_if = "Vec::is_empty", default)
     )]
-    docs: Vec<T::String>,
+    pub(crate) docs: Vec<T::String>,
+}
+
+impl<T> Default for Type<T>
+where
```

Comment:
It would be nice not to have to implement `Default` here where it doesn't make sense. I see that it is required for `mem::take`, I wonder if it is possible to utilize `Option<Type>` in that case

Reply:
I've picked the placeholder approach to avoid changing the types: `Vec<Port>` to `Option<Port>` and avoid an extra unwrap :D Let me know what you think of it Thanks!